UC-Berkeley-Extension/global
# Content moderation

## Asana Ticket(s)
- See ticket [Ticket Title](url).

## Description
This PR builds on #28 to add Content Moderation to the site.

## Testing instructions
1. Build the site
    * From inside the VM, run `/var/www/html/scripts/setup.sh`
1. Go to the Test Article page https://global.ddev.site/node/2
  - Edit the Test Article, set the edit to `Draft` and save
  - Observe you are on the `Latest Version` tab (https://global.ddev.site/node/2/latest)
   - Note that the live version is unchanged (https://global.ddev.site/node/2)
1. Go to the Revisions tab (https://global.ddev.site/node/2/revisions)
  - Compare the two revisions
1. Go to the Edit tab
   - Note that you are editing the *draft* version of the content.
   - Note that you (as admin) can set the content to Draft, In Review, or Published.
   - Set the revision to Published and save.
   - Note that the new version is published and the Latest Version tab is gone.
1. Create a new article
   - Note that the default state is `Draft`


1. Create a user.
1. Login as that user.
1. Create a new article
   - Note that the default state is `Draft`
   - Note that you cannot set the content to `Published`
1. Edit the Article
   - Set the article to `In Review.`

1. As the admin user...
   - Go to https://global.ddev.site/admin/content/moderated and see content under moderation.

## Relevant Screenshots/gifs:

![Screen Shot 2020-10-12 at 11 37 31 AM](https://user-images.githubusercontent.com/360988/95765393-ce82f180-0c7f-11eb-9354-49316ae38aef.png)


## Remaining tasks

- [ ] Review and test with editorial team
- [ ] Edit Workbench views to support moderation state.
